src/platforms/stlink/Makefile.inc

```diff
@@ -25,8 +25,8 @@ else
 LDFLAGS += --specs=nosys.specs
 endif
 
-ifeq ($(SWIM_AS_UART), 1)
-CFLAGS += -DSWIM_AS_UART=1
+ifeq ($(SWIM_NRST_AS_UART), 1)
+CFLAGS += -DSWIM_NRST_AS_UART=1
```

Comment:
> renames build option

How would be newcoming users following old build guides treated here? `make PROBE_HOST=stlink SWIM_AS_UART=1` will silently build the default configuration now, I think. I have a low-priority suggestion to either keep accepting the old/legacy makeflag (possibly with a warning/deprecation message) or outright fail the build. The preprocessor macro value passed to compiler frontends can stay. Meson is probably not affected.

Reply:
Checking if SWIM_AS_UART is set in any capacity and turning that into an error w/ mention of the new option name would be a good idea, but is something we can do as a follow-up if the contributor doesn't wish to do this themself. It's certainly a good point though that we hadn't considered.

Reply:
I can as well add a check for old flag setting new compiler switch, like this:
```
ifeq ($(SWIM_AS_UART), 1)
CFLAGS += -DSWIM_NRST_AS_UART=1
```

and some deprecation warning.

Reply:
That works. We can turn that into a hard error down the road 👍🏼 .